src/stores/AccountController.ts

```diff
@@ -428,38 +453,39 @@ export async function getMediaItems(watchlistId: string | undefined | null, medi
   return getMediaByWatchlist(watchlistId, mediaIds);
 }
 
-async function afterLogin(auth: AuthData, user: Customer, customerConsents: CustomerConsent[] | null, accessModel: string) {
-  useAccountStore.setState({
-    auth,
-    user,
-    customerConsents,
-  });
-
-  return await Promise.allSettled([accessModel === 'SVOD' ? reloadActiveSubscription() : Promise.resolve(), getPublisherConsents()]);
-}
-
-async function getActiveSubscription({ cleengSandbox, customerId, jwt }: { cleengSandbox: boolean; customerId: string; jwt: string }) {
-  const response = await subscriptionService.getSubscriptions({ customerId }, cleengSandbox, jwt);
-
-  if (response.errors.length > 0) return null;
-
-  return response.responseData.items.find((item) => item.status === 'active' || item.status === 'cancelled') || null;
-}
-
-async function getAllTransactions({ cleengSandbox, customerId, jwt }: { cleengSandbox: boolean; customerId: string; jwt: string }) {
-  const response = await getTransactions({ customerId }, cleengSandbox, jwt);
-
-  if (response.errors.length > 0) return null;
+async function afterLogin(
+  auth: AuthData,
+  user: Customer,
+  customerConsents: CustomerConsent[] | null,
+  accessModel: string,
+  shouldSubscriptionReload: boolean = true,
+) {
+  await useService(async ({ accountService }) => {
+    useAccountStore.setState({
+      auth,
+      user,
+      customerConsents,
+    });
 
-  return response.responseData.items;
-}
+    // subscribe to listen to web socket notifications
+    await accountService.subscribeToNotifications(user.uuid, (message) => {
+      const notification = JSON.parse(message);
 
-async function getActivePayment({ cleengSandbox, customerId, jwt }: { cleengSandbox: boolean; customerId: string; jwt: string }) {
-  const response = await getPaymentDetails({ customerId }, cleengSandbox, jwt);
+      notifications[notification.type as NotificationsTypes]?.();
```

Comment:
It's hard to follow what's happening here. I think the lookup is overkill.

Reply:
updated.